qiskit_nature/operators/second_quantization/sparse_label_op.py

```diff
@@ -0,0 +1,102 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The Sparse Label Operator base interface."""
+
+from __future__ import annotations
+from typing import Dict, Iterator
+import math
+from xmlrpc.client import Boolean
+import numpy as np
+
+from qiskit.quantum_info.operators.mixins import LinearMixin, AdjointMixin, TolerancesMixin
+
+
+class SparseLabelOp(LinearMixin, AdjointMixin, TolerancesMixin):
+    """The Sparse Label Operator base interface."""
+
+    def __init__(self, data: Dict[str, complex], register_length: int = None):
+        self._data = data
+        self._register_length = register_length
+
+    def _add(self, other, qargs=None) -> SparseLabelOp:
+        """Return Operator addition of self and other"""
+        new_data = self._data.copy()
+
+        for key, value in other._data.items():
+            if key in new_data.keys():
+                new_data[key] += value
+            else:
+                new_data[key] = value
+
+        return SparseLabelOp(new_data)
+
+    def _multiply(self, other: complex) -> SparseLabelOp:
```

Comment:
```suggestion
    def _multiply(self, other: Number) -> SparseLabelOp:
```
Maybe in combination with `from numbers import Number` this is a bit cleaner?
This could also be used for the `isintance` check below